app/controllers/v0/sessions_controller.rb

```diff
@@ -151,6 +152,19 @@ def log_persisted_session_and_warnings
       end
     end
 
+    # this method is intended to be temporary as we gather more information on the auth_too_late SAML error
+    def log_auth_too_late
+      session_object = Session.find(session[:token])
+      user = User.find(session_object&.uuid)
+
+      log_message_to_sentry('auth_too_late ',
+                            :warn,
+                            code: @sso_service.auth_error_code,
+                            errors: @sso_service.errors.messages,
+                            last_signed_in_if_logged_in: user&.last_signed_in,
+                            authn_context: user&.authn_context)
+    end
```

Comment:
wouldn't it be better to add this to the existing warning we get?

Reply:
I put it in the controller originally when I thought I could just use the `validate` method, but that changed current behavior so I just found the `Session` and `User`. We'd need to start passing `session` if I moved this.

Reply:
ahhh, yes, that makes sense... i guess if its temporary discovery we can live with the duplicated noise.